pkg/commands/build/testdata/testcase-variable-ref/expected.yaml

```diff
@@ -100,10 +94,16 @@ spec:
 apiVersion: v1
 kind: Service
 metadata:
+  annotations:
+    prometheus.io/path: _status/vars
+    prometheus.io/port: "8080"
+    prometheus.io/scrape: "true"
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
   labels:
     app: cockroachdb
-  name: dev-base-cockroachdb-public
+  name: dev-base-cockroachdb
```

Comment:
Why is this order changed? I thought we have solved this ordering problem.

Reply:
All resources are sorted by the return value of `String()` in `ResMap.EncodeAsYaml()`. `ResId.String()` returns the string which concatenated namespace, prefix, name, suffix, etc. with the separater `|`.

```

Reply:
It is caused by the change of sort result. All resources are sorted by the return value of `String()` in `ResMap.EncodeAsYaml()`. `ResId.String()` came to return string with suffix. How should I solve?

https://github.com/kubernetes-sigs/kustomize/pull/550/files#diff-3a1f4dc7c29dd6db0b01c31a739306ffR113

#### before

```
...
"noGroup_v1_Service|noNamespace|dev-:base-|cockroachdb"
"noGroup_v1_Service|noNamespace|dev-:base-|cockroachdb-public"
...
```

#### after

```
...
"noGroup_v1_Service|noNamespace|dev-:base-|cockroachdb-public|noSuffix"
"noGroup_v1_Service|noNamespace|dev-:base-|cockroachdb|noSuffix"
...
```

Reply:
I see, "-" is in front of "|". What I worried about is the output order is random. As long as it's deterministic, I'm fine with that.